zstacklib/zstacklib/test/test_qemu_nbd.py

```
@@ -0,0 +1,149 @@
+            qemu_nbd.os.kill = original_kill
+
+    def test_kill_nbd_fails_when_process_still_exists(self):
+        original_find = qemu_nbd.linux.find_process_list_by_command
+        original_wait = qemu_nbd.linux.wait_callback_success
+        original_kill = qemu_nbd.os.kill
+
+        qemu_nbd.linux.find_process_list_by_command = lambda command, arguments: ['123']
+        qemu_nbd.linux.wait_callback_success = lambda callback, callback_data=None, timeout=60, interval=1: False
+        qemu_nbd.os.kill = lambda pid, sig: None
+        try:
+            self.assertRaises(Exception, qemu_nbd.kill_nbd_process_by_flag, '/dev/vg/volume')
```

Ruff (0.16.0)

[warning] Do not assert blind exception: `Exception`

(B017)